driver_integration_core_test.go

```diff
@@ -0,0 +1,202 @@
+		t.Errorf("results not equal for float Expected: NaN Got: %f", f)
+	}
+
+	if rows.Next() {
+		t.Errorf("Next didn't returned false, although no data is expected")
+	}
+}
+
+// TestDriverInfNanValues tests query with inf and nan values
+func TestDriverInfNanValues(t *testing.T) {
+	db, err := sql.Open("firebolt", dsnMock)
+	if err != nil {
+		t.Errorf("failed unexpectedly with %v", err)
+	}
+	rows, err := db.Query("SELECT '-inf'::double as f, 'inf'::double as f2, 'nan'::double as f3, '-nan'::double as f4")
+	if err != nil {
+		t.Errorf("db.Query returned an error: %v", err)
+	}
+	var f, f2, f3, f4 float64
+
+	if !rows.Next() {
+		t.Errorf("Next returned end of output")
+	}
+	utils.AssertEqual(rows.Scan(&f, &f2, &f3, &f4), nil, t, scanErrorMessage)
+	if !math.IsInf(f, -1) {
+		t.Errorf("results not equal for float Expected: -Inf Got: %f", f)
+	}
+	if !math.IsInf(f2, 1) {
+		t.Errorf("results not equal for float Expected: Inf Got: %f", f2)
+	}
+	if !math.IsNaN(f3) {
+		t.Errorf("results not equal for float Expected: NaN Got: %f", f3)
+	}
+	if !math.IsNaN(f4) {
+		t.Errorf("results not equal for float Expected: NaN Got: %f", f4)
+	}
+}
+
+// TestDriverOpenConnection checks making a connection on opened connector
+func TestDriverOpenConnection(t *testing.T) {
+	db, err := sql.Open("firebolt", dsnMock)
+	if err != nil {
+		t.Errorf("failed unexpectedly")
+	}
+
+	ctx := context.TODO()
+	if _, err = db.Conn(ctx); err != nil {
+		t.Errorf("connection is not established correctly: %v", err)
+	}
+}
+
+// TestDriverOpenEngineUrl checks opening connector with a default engine
+func TestDriverOpenNoDatabase(t *testing.T) {
+	runTestDriverExecStatement(t, dsnNoDatabaseMock)
+}
+
+// TestDriverExecStatement checks exec with full dsn
+func TestDriverExecStatement(t *testing.T) {
+	runTestDriverExecStatement(t, dsnMock)
+}
+
+func TestIncorrectQueryThrowingStructuredError(t *testing.T) {
+	db, err := sql.Open("firebolt", dsnNoDatabaseMock)
+	if err != nil {
+		t.Errorf("failed unexpectedly with %v", err)
+	}
+	_, err = db.Query("SELECT 'blue'::int")
+	if err == nil {
+		t.Errorf("Query didn't return an error, although it should")
+	}
+
+	if !strings.HasPrefix(err.Error(), "query execution error: error during query request:") || !strings.Contains(err.Error(), "Unable to cast text 'blue' to integer") {
+		t.Errorf("Query didn't return an error with correct message, got: %s", err.Error())
+	}
+}
+
+func TestParametrisedQuery(t *testing.T) {
+	ctx := context.TODO()
+	db, err := sql.Open("firebolt", dsnNoDatabaseMock)
+	if err != nil {
+		t.Errorf("failed unexpectedly with %v", err)
+	}
+	query := "SELECT engine_name, status from information_schema.engines WHERE engine_name = ? AND status = ?"
+	stmt, err := db.PrepareContext(ctx, query)
+	if err != nil {
+		t.Errorf("The query %s returned an error: %v", query, err)
+	}
+	rows, err := stmt.QueryContext(ctx, "firebolt-core-engine-name", "RUNNING")
+	//rows, err := stmt.QueryContext(ctx, engineNameMock, "RUNNING")
```

Comment:
Comment.
Also, do we even have i_s.engines in firebolt core? Logically it doesn't make sense.

Reply:
Yes, it contains a single `firebolt-core-engine-name` engine